src/Tribe/Commerce/Cart.php

```diff
@@ -34,15 +34,67 @@ public function process_cart() {
 				$data = stripslashes( $data );
 				$data = json_decode( $data, true );
 			}
+
+			$data = array_merge( $_POST, $data );
 		}
 
-		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? $data['tribe_tickets_post_id'] : null;
-		$provider = isset( $data['tribe_tickets_provider'] ) ? $data['tribe_tickets_provider'] : null;
+		$post_id  = isset( $data['tribe_tickets_post_id'] ) ? absint( $data['tribe_tickets_post_id'] ) : null;
+		$provider = isset( $data['tribe_tickets_provider'] ) ? sanitize_text_field( $data['tribe_tickets_provider'] ) : null;
 		$tickets  = isset( $data['tribe_tickets_tickets'] ) ? $data['tribe_tickets_tickets'] : null;
 		$meta     = isset( $data['tribe_tickets_meta'] ) ? $data['tribe_tickets_meta'] : null;
 
+		if ( null === $provider ) {
+			if ( ! empty( $_GET['provider'] ) ) {
+				$provider = sanitize_text_field( $_GET['provider'] );
+			} elseif ( ! empty( $_POST['provider'] ) ) {
+				$provider = sanitize_text_field( $_POST['provider'] );
+			}
+		}
+
+		$tribe_commerce_providers = [
+			'tpp',
+			'tribe-commerce',
+			'tribe_tpp_attendees',
+			'Tribe__Tickets__Commerce__PayPal__Main',
+		];
+
+		$is_tribe_commerce = in_array( $provider, $tribe_commerce_providers, true );
+
+		// Simplify provider for Tribe Commerce.
+		if ( $is_tribe_commerce ) {
+			$provider = 'tribe-commerce';
```

Comment:
Enforces provider name for tribe-commerce for better uniformity further into Cart handling/hooks